### DISCUSSION TITLE
Is this a bug or an oversight

### DISCUSSION THREAD

**Discussion by thejoyfuldev** [remix-run/remix#2305] [Q&A] [answered]
I was following the blog tutorial and toward the end it directs you to disable JS to show that it still works, and it does. But, it never tells you to re-enable JS because the very next topic is ```useTransition``` which doesn't work unless you have JS enabled.

For some real fun, disable JavaScript in your dev tools and try it out. Because Remix is built on the fundamentals of HTTP and HTML, this whole thing works without JavaScript in the browser. But that's not the point. Let's slow this down and add some "pending UI" to our form.

At this point, a Promise is create to replicate delays and uses the ```useTransition``` hook to update the UI so the user knows something is happening.  While no errors are thrown in the console, this hook does not work without JS enabled.

Am I missing something?

**Comment by machour** [comment]
@jasonabullard you are right, javascript needs to be enabled for the next section.

The sentence just tells you to try disabling JS for fun, and then that this is not the point (so you're supposed to re-enable it).

**Comment by MichaelDeBoey** [comment, marked as answer]
@jasonabullard The blog tutorial is recently changed (see #2502).
Have a look at that one if you're keen to learn.

You can always open a PR to clarify things if you think that's needed.
